compiler/Makefile

```diff
@@ -23,6 +23,19 @@ ifndef CHPL_MAKE_HOME
 export CHPL_MAKE_HOME=$(shell pwd)/..
 endif
 
+# Generate tags only if $TAGS turned on explicitly, or if $CHPL_DEVELOPER is on
```

Comment:
I agree that it'd be ideal to have TAGS be a sibling to OPTIMIZE and DEBUG in terms of where/how it is established.